src/awkward/contents/emptyarray.py

```diff
@@ -109,10 +107,16 @@ def __copy__(self):
     def __deepcopy__(self, memo):
         return self.copy()
 
+    @classmethod
+    def _enforce_parameters(cls, parameters, instance_name):
+        if not (parameters is UNSET or parameters is None or len(parameters) == 0):
+            raise TypeError(
+                f"{instance_name} does not support parameters (given {parameters})"
+            )
+
```

Comment:
Following on the above.

Reply:
That's a fair point. I'll try to back that change out. I added `_enforce_parameters` earlier when I was adding "is_empty_array" to node_parameters, at which point _enforce_parameters was complicated enough to not want to just copy+paste. Then when removing that stuff, lazy-me felt "Eh, this can be argued to be an improvement" even if only barely.

I really don't feel strongly about this one way or another, but the counter-argument to your point is to adopt a kind of policy that any code changes that are thought to be cleanup or simplification are encouraged to happen surrounding any code that one revisits and studies in making a PR. That's a loose policy that sacrifices succinct PRs for gains in code polish -- allowing changes that are too minor to want to formalize into an issue and PR.

Whether or not that sounds reasonable, is what I've done here specifically an improvement or is it noise? It's not obvious to me either way, and I won't argue for it.